### ISSUE TITLE
Warrior Slam Casting

### ISSUE THREAD

**Issue by Manpower99** [gamefreedomgit/Frostmourne#1449]
[//]: # (REMBEMBER! Add links to things related to the bug using for example:)
[//]: # (https://www.wowhead.com/wotlk)

**Description:**
[Patch 3.0.2](https://wowpedia.fandom.com/wiki/Patch_3.0.2) (2008-10-14): Slam now suspends the weapon swing timer rather than resetting it.
Currently on whitemane it "works" in a way where even though the swing timer is suspended, the moment the swing timer continues it will discount the Slam's cast time from the swing timer.

While Bloodsurge(Fury Spec) is procced, Instant cast Slams are working correctly, this is an issue only when casting Slam.

This is testable with two talent points of Improved Slam and Untalented.

**How to reproduce:**
Cast Slam with 1.5s Cast Time > Notice how your swing will resolve at -1.5sec of where its supposed to.
Cast Slam with 1s Cast Time >  your swing will resolve at -1.0sec of where its supposed to.
Cast with a 0.5s Cast Time >  your swing will resolve at -0.5sec of where its supposed to.

**Easiest Way to test it is this :**
1. Do not Talent Imp. Slam, you can test with the talent too but 1.5s is just more easily noticable.
2. Use a 4.0 Base Attack Speed Weapon and Attack a Target Dummy 
3. At around 2sec left of the swing timer, cast Slam
4. Notice the swing resolves 1.5s prematurely(if tested while untalented)

**_Essentially making this talent useless as long you use your Slam before where the progress of your swing would occur by about 2s-0.5s depending on how many talents invested or not invested._**

Or incredibly broken if you just spam slam and every slam reduces your swing timer progress by 0.5(if fully talented).

So Two Slams(with 2/2) in a row would take off 1s off your swing timer while also doing two slams worth of damage.

You could even cast Slam while dual wielding as Fury and you'll see it affects both mainhand and offhand swing progress, easily noticable if you have Titans Grip talented with two slow weapons.

**How it should work:**
It should suspend your swing timer and your swing timer should continue where it left off the moment the cast is over with no changes on the rest of its progress.

You can see this bug occured even back in Sunwell Frosthold
https://www.youtube.com/watch?v=3z4DYZBf7_E
https://www.youtube.com/watch?v=bBxL5pFrwog
Everytime he slams the swing timer resolves prematurely by the same amount of seconds he would have had lost (0.5s)

Observe on this video how it should work.
https://youtu.be/ota8yNRkcf8?si=9WnZQ_eYjhKp9pUM
you can see everytime he slams, his swing timer is properly suspended and when it is finished casting THEN his swing timer continues progressing all the way to 0 instead of 0.5 before it actually hits the target.

Tldr: Casting Slam should pause the swing's progress until the cast is done. currently it "suspends" it by just preventing the white hit from happening but the swing timer keeps going while casting it.

**Database links:**

**Comment by UltraNix** [comment]
Fixed.
